cmake/modules/FindTinyXML2.cmake

```diff
@@ -57,46 +57,58 @@ if(NOT TARGET tinyxml2::tinyxml2)
       BUILD_DEP_TARGET()
     else()
       # This is the fallback case where linux distro's dont ship cmake config files
-      # use the old find_library way
-
-      if(PKG_CONFIG_FOUND)
-        pkg_check_modules(PC_TINYXML2 tinyxml2 QUIET)
-      endif()
-
-      find_path(TINYXML2_INCLUDE_DIR tinyxml2.h
-                                    PATHS ${PC_TINYXML2_INCLUDEDIR})
-      find_library(TINYXML2_LIBRARY_RELEASE NAMES tinyxml2
+      # use the old find_library way. Only do this if we didnt find a cmake config 
+      # in the event of the version < depends version
+      if(NOT TARGET tinyxml2::tinyxml2)
+        if(PKG_CONFIG_FOUND)
+          pkg_check_modules(PC_TINYXML2 tinyxml2 QUIET)
+        endif()
+
+        find_path(TINYXML2_INCLUDE_DIR tinyxml2.h
+                                      PATHS ${PC_TINYXML2_INCLUDEDIR})
+        find_library(TINYXML2_LIBRARY_RELEASE NAMES tinyxml2
+                                             PATHS ${PC_TINYXML2_LIBDIR})
+        find_library(TINYXML2_LIBRARY_DEBUG NAMES tinyxml2d
```

Comment:
Is this for windows? Do we explicitly need to look for a debug version?

Reply:
It's a build option in tinyxml2 that uses a postfix for the debug config type. Stuffs me with what to expect in the Linux world, cover all bases because someone's gonna come back and complain at me eventually